# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/nsIQuotaManagerService.idl
# Commit: 46a5babaafd2
# Full Hash: 46a5babaafd2c0f2198f353d62254cf518dab391
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-15 04:16:11
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Implement initialization of entire persistent storage; r=dom-storage-reviewers,jari
#   
#   IndexedDB still calls QuotaManager::EnsurePersistentOriginIsInitializedInternal
#   from its idle database maintenance. However, we want to get rid of that so the
#   method can be made private. The idle database maintenance should only rely on
# ==============================================================================

diff -r 2ff5f3aaf841 -r 46a5babaafd2 dom/quota/nsIQuotaManagerService.idl
--- a/dom/quota/nsIQuotaManagerService.idl	Mon Oct 14 10:00:26 2024 +0000
+++ b/dom/quota/nsIQuotaManagerService.idl	Mon Oct 14 10:06:14 2024 +0000
@@ -34,6 +34,15 @@
   storageInitialized();
 
   /**
+   * Check if persistent storage is initialized.
+   *
+   * If the dom.quotaManager.testing preference is not true the call will be
+   * a no-op.
+   */
+  [must_use] nsIQuotaRequest
+  persistentStorageInitialized();
+
+  /**
    * Check if temporary storage is initialized.
    *
    * If the dom.quotaManager.testing preference is not true the call will be
@@ -81,6 +90,16 @@
   init();
 
   /**
+   * Initializes persistent storage. This can be used in tests to verify
+   * persistent storage initialization.
+   *
+   * If the dom.quotaManager.testing preference is not true the call will be
+   * a no-op.
+   */
+  [must_use] nsIQuotaRequest
+  initializePersistentStorage();
+
+  /**
    * Initializes temporary storage. This can be used in tests to verify
    * temporary storage initialization.
    *